include/odp/api/spec/schedule.h

```diff
@@ -368,6 +368,26 @@ void odp_schedule_order_lock(unsigned lock_index);
 void odp_schedule_order_unlock(unsigned lock_index);
 
 /**
+ * Release ordered context lock and acquires new lock
+ *
+ * This call is valid only when holding an ordered synchronization context.
+ * Release a previously locked ordered context lock and acquires
+ * a new ordered context lock.
+ * This call is valid only when there is a single ordered context lock and
+ * is invalid if there are nested ordered context lock.
```

Comment:
Yes, the restriction is exactly as you have explained.
I will provide more documentation in the next version. The common use-case we have seen involves using ordered lock to regulate the packet flow and it usually does not involve nested atomic locks and also there is a HW limitation in holding multiple ordered locks.

These changes are mainly for performance optimisation since the same behaviour exposed using this API could be achieved by using two different ODP APIs but by having a single API implementation can better optimise this use-case.